integration_tests/integration_test.go

```diff
@@ -260,15 +261,200 @@ var _ = Describe("Tablecloth HTTP listener", func() {
 		})
 
 	})
+
+	Describe("handling failures starting the new version", func() {
+		var cwd string
+		BeforeEach(func() {
+			cwd, _ = os.Getwd()
+			Expect(os.Symlink(cwd+"/test_servers/v1", cwd+"/test_servers/current")).To(Succeed())
+			Expect(os.Chdir(cwd + "/test_servers/current")).To(Succeed())
+			serverCmd = startServer("./server", "-workingDir="+cwd+"/test_servers/current")
+		})
+		AfterEach(func() {
+			os.Chdir(cwd)
+			os.Remove(cwd + "/test_servers/current")
+		})
+
+		Context("the new server fails to start", func() {
+			It("should continue running the old server", func() {
+				resp, err := http.Get("http://127.0.0.1:8081/")
+				Expect(err).NotTo(HaveOccurred())
+				Expect(resp.StatusCode).To(Equal(200))
+
+				err = os.Remove(cwd + "/test_servers/current")
+				Expect(err).NotTo(HaveOccurred())
+
+				// Non-existent directory will cause starting the server to error
+				err = os.Symlink(cwd+"/test_servers/non_existent", cwd+"/test_servers/current")
+				Expect(err).NotTo(HaveOccurred())
+
+				withSilentOutput(func() {
+					reloadServer(serverCmd)
+				})
+
+				resp, err = http.Get("http://127.0.0.1:8081/")
+				Expect(err).NotTo(HaveOccurred())
+				Expect(resp.StatusCode).To(Equal(200))
```

Comment:
It might be nice to assert that it's still the original version of the code that responds to the HTTP request, for documentation purposes if nothing else.

E.g.:

```
Expect(string(body)).To(ContainSubstring("Hello from v1"))
```

...as you have done for another test below.